course-instructions.md

```diff
@@ -0,0 +1,760 @@
+
+To depend on an input file, you first need to set up a simple target whose command returns the filepath of said file. Like so:
+
+```
+tar_target(my_metadata_csv, "1_fetch/in/my_metadata.csv", format = "file")
+```
+
+Now say you had a function that needed this metadata input file for plotting because it contains latitude and longitude for your sites. To depend on this file as input, do this:
+```
+tar_target(map_of_sites, make_a_map(metadata_file = my_metadata_csv))
+```
+
+but *NOT* like this (this would be the method that doesn't track changes to the file!):
+```
+tar_target(map_of_sites, make_a_map(metadata_file = "1_fetch/in/my_metadata.csv"))
+```
+
+As a general rule, do not put filepaths in the body of a function. :end:
+
+<hr>
+
+### How to depend on a directory for changes
+
+You might have a project where there is a directory :file_folder: with a collection of files. To simplify the example, assume all of the files are `.csv` and have the same format. As part of the hypothetical project goals, these files need to be combined and formatted into a single plottable data.frame. 
+
+In a data pipeline, we'd want assurance that any time the number of files changes, we'd rebuild the resulting data.frame. Likewise, if at any point the _contents_ of any one of the files changes, we'd also want to re-build the data.frame. 
+
+This hypothetical example could be coded as 
+
+```r
+library(targets)
+source("combine_files.R")
+
+list(
+  tar_target(in_files, 
+             c('1_fetch/in/file1.csv',
+               '1_fetch/in/file2.csv',
+               '1_fetch/in/file3.csv'), 
+             format = "file"),
+  tar_target(
+    plot_data, 
+    combine_into_df(in_files)
+  ),
+  tar_target(figure_1_png, my_plot(plot_data))
+)
+```
+
+While this solution would work, it is less than ideal because it doesn't scale well to **many** files, nor would it adapt to new files coming into the `1_fetch/in` directory :file_folder: (the pipeline coder would need to manually add file names to the `in_files` target).
+
+---
+
+Lucky for us, the `targets` package can handle having a directory as a target. If you add a target for a directory, the pipeline will track changes to the directory and will rebuild if it detects changes to the contents a file, the name of a file, or the number of files in the directory changes. 
```

Comment:
I don't use this enough

Reply:
I'll be honest - I am skeptical of this target! I basically _always_ want to know what is going on with _every_ file - is that overkill? I just feel like I've been burned by this before. I'll give this a second look.

Reply:
#29